File: 680-valid-palindrome-ii/valid-palindrome-ii.java
class Solution {
    public boolean isPail(String s, int left, int right){
        while(left<right){
            if(s.charAt(left) != s.charAt(right)){
                return false;
            }
            left++;
            right--;
        }
        return true;
    }
    public boolean validPalindrome(String s) {
        
        int left = 0;
        int right = s.length()-1;
        int count = 0;

        while(left < right){
            if(s.charAt(left) != s.charAt(right)){
               return isPail(s, left+1, right) ||  isPail(s, left, right-1);
            }
            left++;
            right--;
        }
        return true;
    }
}

File: 680-valid-palindrome-ii/valid-palindrome-ii.py
class Solution(object):
    def isPail(self, s, left, right):
        while left < right:
            if s[left] != s[right]:
                return False
            left += 1
            right -= 1

        return True


    def validPalindrome(self, s):

        left = 0
        right = len(s) - 1

        while left < right:

            if s[left] != s[right]:
                return self.isPail(s, left + 1, right) or self.isPail(s, left, right - 1)
    
            left += 1
            right -= 1

        return True
        